### DISCUSSION TITLE
so-firewall add rule issue

### DISCUSSION THREAD

**Discussion by intsec** [Security-Onion-Solutions/securityonion#3577] [Unsupported Versions]
So I have created a firewall rule using so-firewall and following the example on the documentation.  Wheyn I run salt to apply the rule I get this error...

No minions matched the target. No command was sent, no jid was assigned.

I am not sure what I am missing.  Any help would be much appreciated.

Eric

**Comment by intsec** [comment]
Any thoughts?  I would like to be able to connect to kibana from an external source.

**Reply by weslambert** [reply]
Could you give an example of the command(s) you ran?

**Reply by intsec** [reply]
Sure, I used the example commands in the firewall page and created a group name aws using the command "sudo so-firewall addhostgroup aws" then the IP's to the group "sudo so-firewall includehost aws 1.2.3.4" the the creation of the pillar file in /opt/so/saltstack/local/pillar/minions/ using the filename aws_standalone.sls and the used the example file as my template and changed the hostgroup name and portgroup name to portgroups.kibana.  Last was the command to create everything I did "salt aws.standalone state.apply firewall.  Then I get the error "No minions matched the target. No command was sent, no jid was assigned."  Thanks Wes for the help you give!

**Reply by intsec** [reply]
I am finding now that when I create a local suricata rule and run the salt state.apply command to it I am getting the same error back

**Reply by intsec** [reply]
I figured out the issue with the local suricata rule.  When I added my $hostname_$role I didn't put an underscore, I put a period.  That is working but not the firewall rule.

**Reply by intsec** [reply]
And I Just figured the firewall rule issue.  Where it said "salt $Hostname_$Role state.apply firewall", I thought hostname was the hostgroup I created and that hostgroup was a new sls hostname file I created with the rules and I just read that I need to add the rules to the hostname_role.sls original file for the manager.  If that makes any sense.  Anyway, I just needed to reread it a few times to have it click in.

**Comment by intsec** [comment]
My last comment was the answer.  Thanks for the help guys, can't wait to use more of the new version.
